hnn_core/pyramidal.py

```diff
@@ -872,24 +872,35 @@ def parconnect(self, gid, gid_dict, pos_dict, p):
                       'L2_basket', 'L2Basket', lamtha=50.,
                       postsyns=[self.apicaltuft_gabaa])
 
-    # receive from common inputs
     # XXX check NetCon connections for proximal inputs with zero weights
-    def parreceive(self, gid, gid_dict, pos_dict, p_ext):
-        for gid_src, p_src, pos in zip(gid_dict['common'],
-                                       p_ext, pos_dict['common']):
+    def parreceive(self, gids, pos_dict, feeds, threshold):
```

Comment:
`gid` was not used anywhere in the function and `gid_dict` is very opaque. All it does is provide a range of `gids`. This should be passed directly.